modules/core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/jdbc/JDBCMapper.java

```diff
@@ -428,7 +430,7 @@ public IterableQueryResult queryAndFetch(String query, String queryType, QueryFi
         try {
             return new ResultSetQueryResult(queryMaker, query, queryFilter, pathResolver, this, params);
         } catch (SQLException e) {
-            throw new NuxeoException("Invalid query: " + queryType + ": " + query, e);
+            throw new NuxeoException("Invalid query: " + queryType + ": " + query, e, SC_BAD_REQUEST);
```

Comment:
All changes in this class not needed after the constructor changes are done.

Reply:
Hm sorry forget about this, I was wrong.

Reply:
It's a NuxeoException not a QueryParseException, so by default it throws a 500 status code if not precised otherwise.